accumulator/src/wasm.rs

```diff
@@ -0,0 +1,149 @@
+#![allow(missing_copy_implementations)]
+#![allow(clippy::unused_unit)]
+
+macro_rules! export_tree {
+    ($depth:literal) => {
+        affix::paste! {
+            mod [<internal_ $depth>] {
+                use wasm_bindgen::prelude::*;
+                use crate::{Merkle, MerkleProof};
+
+                #[wasm_bindgen(inspectable)]
+                #[derive(Debug, Default, PartialEq)]
+                #[doc = "A sparse merkle tree of depth " $depth]
+                pub struct [<Tree $depth>](pub(crate) crate::Tree<$depth>);
+
+                #[wasm_bindgen(inspectable)]
+                #[derive(Debug, Clone, PartialEq)]
+                #[doc = "A merkle proof of depth " $depth]
+                pub struct [<Proof $depth>](pub(crate) crate::Proof<$depth>);
+
+                type Internal = crate::Tree<$depth>;
+                type InternalProof = crate::Proof<$depth>;
+
+                impl From<InternalProof> for [<Proof $depth>]{
+                    fn from(p: InternalProof) -> [<Proof $depth>]{
+                        [<Proof $depth>](p)
+                    }
+                }
+
+                impl From<Internal> for [<Tree $depth>] {
+                    fn from(p: Internal) -> [<Tree $depth>] {
+                        [<Tree $depth>](p)
+                    }
+                }
+
+                #[wasm_bindgen]
+                impl [<Tree $depth>] {
```

Comment:
Don't we want to `impl Merkle for [<Tree $depth>]` since trees are no trait-based?